app/services/service_listeners/publishing_api_html_attachments.rb

```diff
@@ -18,8 +18,20 @@ def publish
     alias_method :unwithdraw, :publish
 
     def republish
-      update_publishing_api_content
-      unpublish_if_required
+      if Edition::PRE_PUBLICATION_STATES.include?(edition.state)
+        update_draft(update_type: "republish")
+      elsif edition.unpublishing && edition.withdrawn?
+        do_publish("republish")
+        discard_drafts(deleted_html_attachments)
+        withdraw
+      elsif edition.unpublishing
+        update_draft(update_type: "republish")
```

Comment:
What would happen if we didn't update the draft here? Strictly speaking it doesn't feel to me that it should be necessary, but possibly the unpublish request might fail because `allow_draft` is set to `true`?

Reply:
hmm i was thinking strictly from the point of view of keeping whitehall and publishing API up to date. additionally, if there were changes to the draft, surely we would want to see it reflected in draft origin?

Reply:
Oh, yes, that makes sense. Thanks for explaining